sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/DataBatchTests.cs

```diff
@@ -75,30 +75,33 @@ public async Task SendWithPartitionKey()
         [DisplayTestMethodName]
         public async Task SendingPartitionKeyBatchOnPartitionSenderShouldFail()
         {
-            var ehClient = EventHubClient.CreateFromConnectionString(TestUtility.EventHubsConnectionString);
-            var partitionSender = ehClient.CreatePartitionSender("0");
-            try
+            await using (var scope = await EventHubScope.CreateAsync(1))
             {
-                var batchOptions = new BatchOptions()
+                var connectionString = TestUtility.BuildEventHubsConnectionString(scope.EventHubName);
+                var ehClient = EventHubClient.CreateFromConnectionString(connectionString);
+                var partitionSender = ehClient.CreatePartitionSender("0");
```

Comment:
>ehClient.CreatePartitionSender("0"); [](start = 38, length = 36)

I think you can randomize the partition picked.

Reply:
@tzhanl let's create an issue to track this additional improvement!

Reply:
created issue https://github.com/Azure/azure-sdk-for-net/issues/7178 to follow this.